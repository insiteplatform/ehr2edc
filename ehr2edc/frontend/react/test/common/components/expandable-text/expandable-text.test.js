import React from 'react';
import ExpandableText from "../../../../src/common/components/expandable-text/expandable-text";

describe("Rendering the component", () => {
    test("Does not render without children", () => {
        let expandableText = shallow(<ExpandableText/>);

        expect(expandableText).toEqual({});
    });
    test("Renders in closed state without props", () => {
        let expandableText = shallow(<ExpandableText>
            <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab accusamus alias atque deleniti eius
                eligendi, fugit labore, libero nobis odio odit officia temporibus unde. Blanditiis debitis et labore
                soluta totam?</p>
        </ExpandableText>);

        assertCollapsed(expandableText)
        expect(expandableText).toMatchSnapshot();
    });
    test("Renders in closed state with expanded=false", () => {
        let expandableText = renderWithExpanded(false);

        assertCollapsed(expandableText);
        expect(expandableText).toMatchSnapshot();
    });
    test("Renders in open state with expanded=true", () => {
        let expandableText = renderWithExpanded(true);

        assertExpanded(expandableText);
        expect(expandableText).toMatchSnapshot();
    });
});

describe("Expanding and collapsing the component", () => {
    test("Expands a closed component when the action trigger is clicked", () => {
        //given
        let expandableText = renderWithExpanded(false);
        assertCollapsed(expandableText);
        // when
        clickActionTrigger(expandableText);
        //then
        assertExpanded(expandableText);
    });
    test("Expands a closed component when the action trigger is clicked", () => {
        // given
        let expandableText = renderWithExpanded(true);
        assertExpanded(expandableText);
        //when
        clickActionTrigger(expandableText);
        //then
        assertCollapsed(expandableText);
    });
});

function renderWithExpanded(expanded) {
    return shallow(<ExpandableText expanded={expanded}>
        <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab accusamus alias atque deleniti eius
            eligendi, fugit labore, libero nobis odio odit officia temporibus unde. Blanditiis debitis et labore
            soluta totam?</p>
    </ExpandableText>);
}

function assertExpanded(expandableText) {
    expect(expandableText.state("expanded")).toBe(true);
    expect(expandableText.first().hasClass("expanded")).toBe(true);
    expect(expandableText.find("[data-enzyme='action']").text()).toBe("Show less");
}

function assertCollapsed(expandableText) {
    expect(expandableText.state("expanded")).toBe(false);
    expect(expandableText.first().hasClass("expanded")).toBe(false);
    expect(expandableText.find("[data-enzyme='action']").text()).toBe("Show more");
}

function clickActionTrigger(expandableText) {
    expandableText.find("[data-enzyme='action']").simulate('click');
}
