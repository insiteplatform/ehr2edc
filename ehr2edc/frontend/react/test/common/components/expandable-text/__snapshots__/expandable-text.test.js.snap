// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering the component Renders in closed state with expanded=false 1`] = `
<div
  className="expandable-text"
>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab accusamus alias atque deleniti eius eligendi, fugit labore, libero nobis odio odit officia temporibus unde. Blanditiis debitis et labore soluta totam?
  </p>
  <p>
    <a
      data-enzyme="action"
      onClick={[Function]}
    >
      Show more
    </a>
  </p>
</div>
`;

exports[`Rendering the component Renders in closed state without props 1`] = `
<div
  className="expandable-text"
>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab accusamus alias atque deleniti eius eligendi, fugit labore, libero nobis odio odit officia temporibus unde. Blanditiis debitis et labore soluta totam?
  </p>
  <p>
    <a
      data-enzyme="action"
      onClick={[Function]}
    >
      Show more
    </a>
  </p>
</div>
`;

exports[`Rendering the component Renders in open state with expanded=true 1`] = `
<div
  className="expandable-text expanded"
>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab accusamus alias atque deleniti eius eligendi, fugit labore, libero nobis odio odit officia temporibus unde. Blanditiis debitis et labore soluta totam?
  </p>
  <p>
    <a
      data-enzyme="action"
      onClick={[Function]}
    >
      Show less
    </a>
  </p>
</div>
`;
